### ISSUE TITLE
cc3100-dev: To fix before making pull request

### ISSUE THREAD

**Issue by blmorris** [blmorris/micropython#1]
Issues that I have found while working with the cc3100 driver:
- [ ] Cannot connect to WEP network (can connect to WPA2, nto tested WPA)
- [ ] Class initialization accepts pin arguments, but does not use them - connection is hard-coded to SPI(2), Y5, Y4, and Y3
- [ ] Change class name from CC31K to CC3K1 (or to something else; I believe the CC3K1 pattern was suggested elsewhere in the MicroPython discussions but can't find it now.)

Feel free to add others.

**Comment by danicampora** [comment]
WEP doesn't work on the CC3200 either. I think is related to the fact that in only supports WEP in HEX mode. I am going to try to add a fix for that now.
WPA/WPA2 and Open networks connect just fine.

**Comment by blmorris** [comment]
Thanks, that's good to know. All the more reason to support the CC3100 with one driver across both ports; fix it once and it's done.

**Comment by danicampora** [comment]
@blmorris this: https://github.com/micropython/micropython/commit/92ea99a0fb28ecfdf81f9af9456fa2b423d498cd fixes the problem with WEP. Please note that WEP secured networks use an hex key. The simplelink library needs the key for WEP networks to be passed as binary, not as a string like done for WPA/WPA2. Check the example at the end of this wiki: http://processors.wiki.ti.com/index.php/CC31xx_Programmers_Guide

**Comment by danicampora** [comment]
True. Now we need to merge the CC3200 WLAN module and the CC3k1 into one.

**Comment by blmorris** [comment]
@danicampora - as you work on merging Nadim's version of the cc3100 driver with yours, I just wanted to call your attention to the changes that were made to modnetwork.[ch] and modusocket.c in addition to the changes to Makefile and qstrdefsport.h,
I realized that those changes might have gotten lost in the mass of changes made by committing everything at once.

**Comment by danicampora** [comment]
Thanks, I will pay attention to that.
